packages/expo-app-integrity/android/src/main/java/expo/modules/integrity/IntegrityModule.kt

```diff
@@ -86,6 +95,33 @@ class IntegrityModule : Module() {
         }
       )
     }
+
+    AsyncFunction("isHardwareAttestationSupported") { promise: Promise ->
+      try {
+        val isSupported = isHardwareAttestationSupported()
+        promise.resolve(isSupported)
+      } catch (e: Exception) {
+        promise.reject(IntegrityException(IntegrityErrorCodes.HARDWARE_ATTESTATION_NOT_SUPPORTED, e.message ?: "Failed to check hardware attestation support", e))
+      }
+    }
```

Comment:
```suggestion
    AsyncFunction("isHardwareAttestationSupported") { 
      try {
        val isSupported = isHardwareAttestationSupported()
        return isSupported
      } catch (e: Exception) {
        throw IntegrityException(IntegrityErrorCodes.HARDWARE_ATTESTATION_NOT_SUPPORTED, e.message ?: "Failed to check hardware attestation support", e)
      }
    }
```
Could this be sync?

Reply:
actually it can [throw](https://github.com/expo/expo/pull/39725/files#diff-0691d9747109d1ac55fa663737af0b0c9ed5ab671d7f7b526273fb65f1ad9f2aR185) so not much useful to keep it sync right? people cant directly use it without try/catch. I am currently in favor of throwing an exception here, in case someone wants to check why the `AndroidKeyStore` was not accessible 🤔

Reply:
That's fine, it didn't throw when I looked at it at first 👍